runtime/hsa-runtime/core/runtime/amd_aie_aql_queue.cpp

```diff
@@ -41,28 +41,51 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #include "core/inc/amd_aie_aql_queue.h"
+#include "core/inc/amd_xdna_driver.h"
 
 #ifdef __linux__
 #include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
-#include <sys/syscall.h>
-#include <unistd.h>
+#include <sys/ioctl.h>
 #endif
 
 #ifdef _WIN32
 #include <Windows.h>
 #endif
 
-#include <stdio.h>
-#include <string.h>
-#include <thread>
+#include <cstring>
 
 #include "core/inc/queue.h"
 #include "core/inc/runtime.h"
 #include "core/inc/signal.h"
 #include "core/util/utils.h"
 
+// The number of arguments in the packet payload before we start passing operands
+constexpr int NON_OPERAND_COUNT = 6;
+
+// Used to transform an address into a device address
+constexpr int DEV_ADDR_BASE = 0x04000000;
+constexpr int DEV_ADDR_OFFSET_MASK = 0x02FFFFFF;
+
+// The driver places a structure before each command in a command chain.
+// Need to increase the size of the command by the size of this structure.
+// In the following xdna driver source can see where this is implemented:
+// Commit hash: eddd92c0f61592c576a500f16efa24eb23667c23
+// https://github.com/amd/xdna-driver/blob/main/src/driver/amdxdna/aie2_msg_priv.h#L387-L391
+// https://github.com/amd/xdna-driver/blob/main/src/driver/amdxdna/aie2_message.c#L637
+constexpr int CMD_COUNT_SIZE_INCREASE = 3;
+
+// Index of command payload where the instruction sequence
+// address is located
+constexpr int CMD_PKT_PAYLOAD_INSTRUCTION_SEQUENCE_IDX = 2;
+
+// Environment variable to define job submission timeout
+constexpr const char *TIMEOUT_ENV_VAR = "ROCR_AIE_TIMEOUT";
+constexpr int DEFAULT_TIMEOUT_VAL = 50;
+char *timeout_env_var_ptr = getenv(TIMEOUT_ENV_VAR);
```

Comment:
Maybe this should be done during driver init?

Reply:
Yep good idea!